### PR TITLE
support RuntimeField with no script

### LINKED ISSUE #2267: RuntimeField with no script
Should allow creating `RuntimeField` with no script.

Usage:
Cast flattened field type from keyword to double.
Mapping:
```
"mappings": {
    "_doc": {
      "properties": {
        "myprops": {
          "type": "flattened"
```
Es query with:
```
    "runtime_mappings": {
        "myprops.myprop": {
            "type": "double"
        }
    }
```

### PR DESCRIPTION
**Pull request by cdalexndr**
- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] **There is a ticket in the bug tracker for the project in our [issue tracker](https://github.
  com/spring-projects/spring-data-elasticsearch/issues)**. Add the issue number to the _Closes #issue-number_ line below
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

Closes #2267

### CODE REVIEW

File: src/test/java/org/springframework/data/elasticsearch/core/RuntimeFieldTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package org.springframework.data.elasticsearch.core;
+
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
```
Comment (xhaggi) [inline src/test/java/org/springframework/data/elasticsearch/core/RuntimeFieldTest.java:5]: We use [AssertJ](https://assertj.github.io/) for assertions.

```java
import static org.assertj.core.api.Assertions.*;
```

File: src/test/java/org/springframework/data/elasticsearch/core/RuntimeFieldTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ */
+class RuntimeFieldTest {
+
+	@Test
+	void testMapping() {
+		RuntimeField runtimeField = new RuntimeField("name", "double", "myscript");
+		Map<String, Object> mapping = runtimeField.getMapping();
+		assertEquals("double", mapping.get("type"));
+		assertEquals("myscript", mapping.get("script"));
+	}
```
Comment (xhaggi) [inline src/test/java/org/springframework/data/elasticsearch/core/RuntimeFieldTest.java:22]: and here:

```java

@Test // #2267
@DisplayName("should return mapping with all properties")
void shouldReturnMappingWithAllProperties() {

  RuntimeField runtimeField = new RuntimeField("name", "double", "myscript");
  Map<String, Object> mapping = runtimeField.getMapping();

  assertThat(mapping).containsEntry("type", "double");
  assertThat(mapping).containsEntry("script", "myscript");
}
```

File: src/test/java/org/springframework/data/elasticsearch/core/RuntimeFieldTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+	@Test
+	void testMappingNoScript() {
+		RuntimeField runtimeField = new RuntimeField("name", "double");
+		Map<String, Object> mapping = runtimeField.getMapping();
+		assertEquals("double", mapping.get("type"));
+		assertFalse(mapping.containsKey("script"));
+	}
```
Comment (xhaggi) [inline src/test/java/org/springframework/data/elasticsearch/core/RuntimeFieldTest.java:30]: and here:

```java
@Test // #2267
@DisplayName("should return mapping without script properties")
void shouldReturnMappingWithoutScriptProperty() {

  RuntimeField runtimeField = new RuntimeField("name", "double");
  Map<String, Object> mapping = runtimeField.getMapping();

  assertThat(mapping).containsEntry("type", "double");
  assertThat(mapping).doesNotContainKey("script");
}
```

**Review by xhaggi — CHANGES_REQUESTED** [review]

File: src/test/java/org/springframework/data/elasticsearch/core/RuntimeFieldTest.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import java.util.Map;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author cdalxndr
+ */
+class RuntimeFieldTest {
+
+	@Test
```
Comment (sothawo) [inline src/test/java/org/springframework/data/elasticsearch/core/RuntimeFieldTest.java:16]:
```suggestion
	@Test // @issue number
```
reference the issue

**Review by sothawo — CHANGES_REQUESTED** [review]

**Pull request merged**